### ISSUE TITLE
Strange assert when using HTTPS on REST API service

### ISSUE THREAD

**Issue by tchaloupka** [vibe-d/eventcore#33]
This is a backtrace from the GDB:

```D
#0  0x0000000001382444 in _d_assert_msg ()
#1  0x00000000012742de in _D9eventcore7drivers5posix7sockets__T23PosixEventDriverSocketsHTCQClQCeQBz5epoll14EpollEventLoopZQCi10releaseRefMFNbNfSQEn6driver__T6HandleVAyaa6_736f636b6574TSQGcQBp__TQBlVQBha2_6664TmVmN1ZQCgVQBjS1N1ZQCsZb (this=0x7ffff7ef3b40, fd=...) at /home/tomas/.dub/packages/eventcore-0.8.22/eventcore/source/eventcore/drivers/posix/sockets.d:776
#2  0x0000000001237005 in vibe.core.net.TCPConnection.~this() (this=...) at /home/tomas/.dub/packages/vibe-core-1.3.0-rc.1/vibe-core/source/vibe/core/net.d:491
#3  0x00000000011bfc95 in _D6object__T16_destructRecurseTS4vibe4core3net13TCPConnectionZQByFNbNfKQBoZv (s=...) at /usr/include/dmd/druntime/import/object.d:2489
#4  0x00000000011bfbf1 in _D6object__T7destroyTS4vibe4core3net13TCPConnectionZQBoFNbNfKQBoZv (obj=...) at /usr/include/dmd/druntime/import/object.d:2860
#5  0x0000000001186d3c in _D4vibe8internal14interfaceproxy__T14InterfaceProxyTCQBz4core6stream6StreamZQBp__T9ProxyImplTSQDoQBp3net13TCPConnectionZQBm8_destroyMFNbNeAvZv (this=0x7fffcf062cc0, stor=...) at /home/tomas/.dub/packages/vibe-core-1.3.0-rc.1/vibe-core/source/vibe/internal/interfaceproxy.d:244
#6  0x000000000117ac91 in _D4vibe8internal14interfaceproxy__T14InterfaceProxyTCQBz4core6stream6StreamZQBp5clearMFNbNfZv (this=...) at /home/tomas/.dub/packages/vibe-core-1.3.0-rc.1/vibe-core/source/vibe/internal/interfaceproxy.d:123
#7  0x000000000117ac21 in _D4vibe8internal14interfaceproxy__T14InterfaceProxyTCQBz4core6stream6StreamZQBp6__dtorMFNbNfZv (this=...) at /home/tomas/.dub/packages/vibe-core-1.3.0-rc.1/vibe-core/source/vibe/internal/interfaceproxy.d:112
#8  0x0000000001210619 in vibe.stream.openssl.OpenSSLStream.__fieldDtor() (this=0x7fffcf049c00) at /home/tomas/.dub/packages/vibe-d-0.8.2-rc.1/vibe-d/tls/vibe/stream/openssl.d:230
#9  0x000000000121063a in vibe.stream.openssl.OpenSSLStream.__aggrDtor() (this=0x7fffcf049c00) at /home/tomas/.dub/packages/vibe-d-0.8.2-rc.1/vibe-d/tls/vibe/stream/openssl.d:230
#10 0x000000000138b50b in rt_finalize2 ()
#11 0x00000000013e17a2 in rt_finalizeFromGC ()
#12 0x00000000013db94e in _D2gc4impl12conservativeQw3Gcx5sweepMFNbZm ()
#13 0x00000000013dc29d in _D2gc4impl12conservativeQw3Gcx11fullcollectMFNbbZm ()
#14 0x00000000013da07e in _D2gc4impl12conservativeQw3Gcx10smallAllocMFNbhKmkZPv ()
#15 0x00000000013de0fd in _D2gc4impl12conservativeQw14ConservativeGC__T9runLockedS_DQCeQCeQCcQCnQBs12mallocNoSyncMFNbmkKmxC8TypeInfoZPvS_DQEgQEgQEeQEp10mallocTimelS_DQFiQFiQFgQFr10numMallocslTmTkTmTxQCzZQFcMFNbKmKkKmKxQDsZQDl ()
#16 0x00000000013d7be3 in _D2gc4impl12conservativeQw14ConservativeGC6qallocMFNbmkxC8TypeInfoZS4core6memory8BlkInfo_ ()
#17 0x00000000013874a7 in gc_qalloc ()
#18 0x00000000013e1384 in _D2rt8lifetime12__arrayAllocFNaNbmxC8TypeInfoxQlZS4core6memory8BlkInfo_ ()
#19 0x000000000138aed2 in _d_newarrayU ()
#20 0x0000000000da27eb in _D6object__T4_dupTaTyaZQlFNaNbAaZAya (a=...) at /usr/include/dmd/druntime/import/object.d:3890
#21 0x0000000000da27c4 in _D6object__T11_trustedDupTaTyaZQtFNaNbNeAaZAya (a=...) at /usr/include/dmd/druntime/import/object.d:3870
#22 0x0000000000da27a0 in _D6object__T4idupTaZQiFNaNbNdNfAaZAya (a=...) at /usr/include/dmd/druntime/import/object.d:3857
#23 0x0000000000fbd794 in _D3std4conv__T6toImplTAyaTAaZQpFNaNbNfQmZQt (value=...) at /usr/include/dmd/phobos/std/conv.d:891
#24 0x00000000010d2d0f in _D3std4conv__T6toImplTAyaTPxaZQqFQhZ9__lambda2MFNaNbZQBf (this=0x7fffd8fee090) at /usr/include/dmd/phobos/std/conv.d:941
```

It happens while reading from the DB. It seems that GC has run during the reading and tries to clean something it shouldn't.
Using latest vibe-0.8.2-rc1, with vibe-core-1.3.0-rc1 on Ubuntu Linux 16.04.03 x86_64

**Comment by s-ludwig** [comment]
There is a good chance that this was fixed by 31286e45808936c857aee968027c3ae59e098884, which makes sure that the socket handle passed to read/write callbacks is valid even if all external references have already been given up. I've tagged 0.8.24, which contains the fix, so you could try to see if it still reproduces now.

**Comment by tchaloupka** [comment]
Ok, havent seen it so far so closing this and will reopen if found again
